### DISCUSSION TITLE
Create script/snippet/clipboard from template

### DISCUSSION THREAD

**Discussion by danielo515** [johnlindquist/kit#1089] [Ideas]
Hello.
I think it will be awesome fi you can select a script template when creating a new script. Right now is very convenient to be able to create a script as quick, but having to start from scratch every time is a bit tedious. For example, when I'm writing templates there is certain boilerplate that is going to be required on every one, same for snippets.
Thanks for this cool tool by the way!

**Comment by danielo515** [comment]
Ok, I just discovered that, somehow, you have categories of things, where you also get the option to create a new one of that type which will indeed use a template.
I find it a bit not intuitive, because there are tabs, but this is like another section that is not present in the tabs... took me a while to discover.
Also, I see there is no category for templates? Are they just supposed to be used in snippets for example?

**Reply by johnlindquist** [reply]
To be honest, you're the first person to ask for that functionality, so I haven't bothered making it a more prominent feature.

**Reply by danielo515** [reply]
Are the templates in templates folder supposed to be used to define this templates? What are the clipboard templates for?

**Reply by johnlindquist** [reply]
@danielo515 

1. "templates folder" - yes, that's the purpose
2. "clipboard templates" - That's a leftover legacy template

"Clipboard templates" were replaced by a `template` helper to build strings that works the same as VS Code's template/snippets

```
let colors = `Red,Green,Blue,Black,White`

let text = await template(`My favorite color is \${1|${colors}|}`)

setSelectedText(text)
```

**Reply by nbbaier** [reply]
I just want to say that I'd also love the ability to use templates to make scripts!

**Reply by johnlindquist** [reply]
@nbbaier If you update the `~/.kenv/templates/default.js` (or `defaults.ts` for TypeScript), then those will be used any time you create a new script. For example, you can add:

```js
// Author: John Lindquist
// Twitter: @johnlindquist 

import "@johnlindquist/kit"
```

So you don't have to add your author/twitter metadata every time
